test/core/end2end/tests/graceful_server_shutdown.cc

```diff
@@ -38,24 +38,29 @@ CORE_END2END_TEST(Http2Test, GracefulServerShutdown) {
       .RecvInitialMetadata(server_initial_metadata)
       .RecvStatusOnClient(server_status);
   auto s = RequestCall(101);
-  Expect(101, true);
-  Step();
-  // shutdown and destroy the server
-  ShutdownServerAndNotify(200);
-  Step();
-  IncomingCloseOnServer client_close;
-  s.NewBatch(102)
-      .SendInitialMetadata({})
-      .SendStatusFromServer(GRPC_STATUS_UNIMPLEMENTED, "xyz", {})
-      .RecvCloseOnServer(client_close);
-  Expect(102, true);
-  Expect(200, true);
-  Expect(1, true);
+  bool got_server = false;
+  Expect(101, Maybe{&got_server});
   Step();
+  if (got_server) {
+    // shutdown and destroy the server
+    ShutdownServerAndNotify(200);
```

Comment:
I don't understand how this will help. In the existing code, we first verify that the server has received the RPC. Then shutdown the server. There is no connection timeout at play here.